### PR TITLE
update perl-yaml to 1.24

### PR DESCRIPTION
**Pull request by mtw**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by nsoranzo** [comment]
@mtw Any specific reason why you added the `1.24/` subdirectory instead of updating the main `recipes/perl-yaml/meta.yaml` ?

File: recipes/perl-yaml/1.24/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - perl >=5.22,<6.0
+    - perl-extutils-makemaker
+    - perl-test-yaml
+    - perl-test-more
+
+  run:
+    - perl >=5.22,<6.0
+
+test:
+  impprts:
```
Comment (nsoranzo) [inline recipes/perl-yaml/1.24/meta.yaml]: This should have been `imports:`

Comment (mtw) [reply recipes/perl-yaml/1.24/meta.yaml]: no special reason for that

Comment (mtw) [reply recipes/perl-yaml/1.24/meta.yaml]: thanks :)

Comment (nsoranzo) [reply recipes/perl-yaml/1.24/meta.yaml]: OK, I think the best practice is to modify the main `meta.yaml`, unless you need to keep an old version around.
Updated to 1.26 in https://github.com/bioconda/bioconda-recipes/pull/9876

**Pull request merged**